File: backend/src/main/java/com/blackopalsolutions/goneviral/backend/model/response/GetCardResponse.java
package com.blackopalsolutions.goneviral.backend.model.response;

import com.blackopalsolutions.goneviral.backend.model.domain.Card;

public class GetCardResponse extends Response {
    private final Card card;

    public GetCardResponse(Card card) {
        super(card != null);
        this.card = card;
    }

    public GetCardResponse(String message) {
        super(message);
        this.card = null;
    }

    public Card getCard() {
        return card;
    }
}

File: shared/src/commonMain/kotlin/com/blackopalsolutions/goneviral/model/response/GetCardResponse.kt
package com.blackopalsolutions.goneviral.model.response

import com.blackopalsolutions.goneviral.model.domain.Card

class GetCardResponse : Response {
    val card: Card?

    constructor(card: Card) : super(true) {
        this.card = card
    }

    constructor(message: String) : super(message) {
        card = null
    }
}